apx_gui/windows/main_window.py

```diff
@@ -135,3 +137,58 @@ def new_pkgmanager(self) -> None:
             self, self.__pkgmanagers
         )
         window.show()
+
+    def import_file(self) -> None:
+        file_picker = Gtk.FileDialog()
+        file_picker.open(parent=self, cancellable=None, callback=self.open_file_callback)
+
+    def open_file_callback(self, filedialog, task):
+        try:
+            file = filedialog.open_finish(task)
+            with open(file.get_path()) as imported_file:
+                contents = yaml.load(imported_file, Loader=yaml.SafeLoader)
+
+                try:
+                    contents["needsudo"]
+                    entity: PkgManager = PkgManager(
+                        contents["name"],
+                        contents["needsudo"],
+                        contents["cmdautoremove"],
+                        contents["cmdclean"],
+                        contents["cmdinstall"],
+                        contents["cmdlist"],
+                        contents["cmdpurge"],
+                        contents["cmdremove"],
+                        contents["cmdsearch"],
+                        contents["cmdshow"],
+                        contents["cmdupdate"],
+                        contents["cmdupgrade"],
+                        False,
+                    )
+                    result = entity.create()
```

Comment:
@mirkobrombin this is mostly working, but ive been getting an error from create despite the apx entity being created.

what i havent gotten working is an async implementation of this, but i think that has more to do with the error in the entitiy creation.  

before i get too far into this, are you good with me proceeding in this feature and in this way?

Reply:
which its running in a callback right now so i guess it is async in nature

Reply:
this may actually work as is, but the version of apx on my system is hitting the buffer issue so its failing to report back successfully